nnet_utils/nnet_conv.h

```diff
@@ -47,6 +47,43 @@ struct conv_config
     static const unsigned n_zeros = 0; // not used yet
 };
 
+
+//Computes multiplier limit
+//This function should not be synthesized into firmware
+template<typename CONFIG_T>
+int compute_multiplier_limit(
+    typename CONFIG_T::weight_t  weights[CONFIG_T::y_filt * CONFIG_T::n_chan * CONFIG_T::n_filt]
+)
+{
+    int n_mult = 0;
+    for(int ii = 0; ii < CONFIG_T::y_out; ii++) {
+        for(int ff = 0; ff < CONFIG_T::n_filt; ff++){
+	    for(int cc = 0; cc < CONFIG_T::n_chan; cc++){
+                for(int jj = 0; jj < CONFIG_T::y_filt; jj++){
+                    
+                    int index_mult   = ii*CONFIG_T::n_filt*CONFIG_T::n_chan*CONFIG_T::y_filt + ff*CONFIG_T::n_chan*CONFIG_T::y_filt + cc*CONFIG_T::y_filt + jj;
+                    int index_weight = jj*CONFIG_T::n_chan*CONFIG_T::n_filt + cc*CONFIG_T::n_filt + ff;
+                    
+                    if((ii*CONFIG_T::stride+jj) < CONFIG_T::pad_left || (ii*CONFIG_T::stride+jj) >= (CONFIG_T::pad_left + CONFIG_T::y_in)){
+			//padded -- do nothing
+			continue;
+                    }
+                    else {
+			//need to tune this cut?
+                        if( weights[index_weight] > 1e-20 || weights[index_weight] < -1e-20 ){
```

Comment:
Do we want to use `np.isclose()` or `math.isclose()` ?
https://stackoverflow.com/questions/5595425/what-is-the-best-way-to-compare-floats-for-almost-equality-in-python

Reply:
It looks like isclose is doing the same thing, just with a configurable tolerance.  From a little googling, I don't know of a function for this in C++ (have you found one?).  However, in your link and the examples I saw, the tolerance was more like 1e-8, not 1e-20.  Maybe we should make ours bigger?

Reply:
this is fine for now